### ISSUE TITLE
`extra_unused_lifetimes` false positive regression from 1.82 to 1.83

### ISSUE THREAD

**Issue by davidhewitt** [rust-lang/rust-clippy#13578]
### Summary

I believe the following to be a regression in a new false positive for the `extra_unused_lifetimes` lint for 1.83. 

The suggestion is to remove a lifetime `'a`, but it is used in the `where` bound for `&'a T: Foo`. It's not valid to use `where &'_ T: Foo` and I also believe `where for<'a> &'a T: Foo` to have a different meaning, so I do not see a way to remove `'a` from the `impl`.

I am of course happy to be wrong, in which case TIL! 😄 

### Reproducer

I tried this code:

```rust
pub trait Foo {}

impl<'a, T: 'a> Foo for Option<T> where &'a T: Foo {}
```

I expected to see this happen: no lint warning. (This is the case on 1.82 stable.)

Instead, this happened:

```
warning: this lifetime isn't used in the impl
 --> src/lib.rs:5:6
  |
5 | impl<'a, T: 'a> Foo for Vec<T> where &'a T: Foo {}
  |      ^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#extra_unused_lifetimes
  = note: `#[warn(clippy::extra_unused_lifetimes)]` on by default
```

I observe this on 1.83.0-beta.2

### Version

rustc 1.83.0-beta.2 (88c1c3c11 2024-10-18)
binary: rustc
commit-hash: 88c1c3c1102bbf3860891bfa52b7ddd9f26aec2f
commit-date: 2024-10-18
host: aarch64-apple-darwin
release: 1.83.0-beta.2
LLVM version: 19.1.1
